### DISCUSSION TITLE
any plan to make these extensions available for Vite

### DISCUSSION THREAD

**Discussion by jianyunli** [quasarframework/quasar#11799] [Quasar App Extensions]
I believe Vue 3 with `<Script setup>`  and Vite and Volar extension is the "recommended" way, I personally adopt it. Is there any plan to make these excellent extensions available to Vite?

**Comment by rstoenescu** [comment]
You can use them already. Nothing to wait for.

**Reply by jianyunli** [reply]
Thank you for reply, that's great news. can you give a hint where can I get info? I am looking at documentation here at https://quasar.dev/app-extensions/introduction

> WARNING

App Extensions are designed specifically for Quasar CLI only. This means that you will not be able to install them or run them with the Quasar Vite plugin or with Vue CLI or in UMD environments.

**Reply by jianyunli** [reply]
oh, I think I got what you mean here, I have to use Quasar CLI to install the extension, but I can use them like standard plugins in the Composition API way, right?

> `app.use(Quasar, {
    plugins: { Dialog, Notify, extensionPlugin }, // import Quasar plugins and add here
    config: {
        notify: { /* look at QuasarConfOptions from the API card */ }
    }
})`

**Reply by hawkeye64** [reply]
If you're talking about app extensions, like QCalendar, then you use the UI package instead of the app-extension package. In this case, you would install `@quasar/quasar-ui-qcalendar` and treat it like any other Vue component in the wild. Most, if not all, app extensions tell you how to install if you are using Vue CLI (and sometimes Vite) on a readme. They are basically the same.

**Reply by jianyunli** [reply]
Thanks again, I am trying to use QGrid
https://next-quasar-qgrid.netlify.app/

looks like it does not have NPM package, the doc says `quasar ext add qgrid` 

I installed Quasar CLI, is this available for my app following composition API with Vite?

**Reply by hawkeye64** [reply]
To use the UI component, the docs says this: 
```
import { Component as QGrid } from 'quasar-ui-qgrid'
```
https://github.com/pratik227/quasar-qgrid/tree/next/ui#vue-cli-project

**Reply by CharlieDigital** [reply]
> Most, if not all, app extensions tell you how to install if you are using Vue CLI (and sometimes Vite) on a readme. They are basically the same.

@hawkeye64 is this also true for `q-pdf-viewer`?

**Reply by hawkeye64** [reply]
@CharlieDigital Actually, that one (QPdfviewer) is only available using Quasar CLI.

**Reply by CharlieDigital** [reply]
That's what I thought as I was going through the code.  I was able to extract a stripped down non CLI version for my project; not sure if there is any interest/value in adding that into the codebase.

**Reply by hawkeye64** [reply]
Codebase? Do you mean Quasar? It won't go in there because PDFjs is quite bloated.

**Reply by CharlieDigital** [reply]
The QPdfviewer extension codebase; not sure if there is interest in adding a non-Quasar CLI alternative into the repo.

Maybe a fork for those that want to run it with Vite?

**Reply by yusufkandemir** [reply]
@hawkeye64, @CharlieDigital probably requests UI kit variant of QPDFViewer, just like with other UI-related AEs. Judging from https://github.com/quasarframework/app-extension-qpdfviewer/tree/next and https://github.com/quasarframework/app-extension-qpdfviewer/tree/next2/ I think it's planned and still WIP.

**Reply by hawkeye64** [reply]
Yeah, we tried and could not get it to work as a UI component. PDFjs is very finicky in the way it works.

**Reply by tombed154** [reply]
@jianyunli Hi im having trouble to make the  QGrid component work. Have you suceeded? If I ry to import it like it says in readme im getting error `The requested module '/node_modules/.vite/deps/quasar-ui-qgrid.js?t=1659087973036&v=7c418584' does not provide an export named 'Component'`

**Comment by pratik227** [comment]
`import QGrid from 'quasar-ui-qgrid/src/components/QGrid.vue'
`
Try this
